mountpoint-s3-client/src/mock_client.rs

```diff
@@ -472,6 +474,24 @@ impl MockObject {
     pub fn etag(&self) -> ETag {
         self.etag.clone()
     }
+
+    pub fn checksum_algorithm(&self) -> Option<ChecksumAlgorithm> {
+        let Checksum {
+            checksum_crc32,
+            checksum_crc32c,
+            checksum_sha1,
+            checksum_sha256,
+        } = &self.checksum;
+
+        match (checksum_crc32, checksum_crc32c, checksum_sha1, checksum_sha256) {
+            (Some(_), None, None, None) => Some(ChecksumAlgorithm::Crc32),
+            (None, Some(_), None, None) => Some(ChecksumAlgorithm::Crc32c),
+            (None, None, Some(_), None) => Some(ChecksumAlgorithm::Sha1),
+            (None, None, None, Some(_)) => Some(ChecksumAlgorithm::Sha256),
+            (None, None, None, None) => None,
+            _ => unreachable!("should not have more than one checksum"),
```

Comment:
Is `unreachable` appropriate here? It seems inconsistent with the comment on the field:
```
    /// Typically, at most one of the checksums should be set.
```

Reply:
Yeah, it is leaving the door open in case the assumption of only one checksum should change but it also means we'd crash here.

I can update the comment to assume there will _never_ be more than one checksum. Alternatively, we can just return the first one set (by replacing `None` with `_` in the patterns).

I'd inclined to do the latter, what do you think?

Reply:
Let's go with the latter.